test/jmap/mail/email/search_snippet/search_snippet_get_method_test.dart

```diff
@@ -0,0 +1,209 @@
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/filter/filter.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/core/request/reference_path.dart';
+import 'package:jmap_dart_client/jmap/core/state.dart';
+import 'package:jmap_dart_client/jmap/jmap_request.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_filter_condition.dart';
+import 'package:jmap_dart_client/jmap/mail/email/query/query_email_method.dart';
+import 'package:jmap_dart_client/jmap/mail/email/search_snippet/search_snippet.dart';
+import 'package:jmap_dart_client/jmap/mail/email/search_snippet/search_snippet_get_method.dart';
+import 'package:jmap_dart_client/jmap/mail/email/search_snippet/search_snippet_get_response.dart';
+
+void main() {
+  final baseOption  = BaseOptions(method: 'POST');
+  final dio = Dio(baseOption)..options.baseUrl = 'http://domain.com/jmap';
+  final dioAdapter = DioAdapter(dio: dio);
+
+  final sessionState = State('some-session-state');
+  final state = State('some-state');
+  final accountId = AccountId(Id('some-account-id'));
+
+  Map<String, dynamic> generateResponse({
+    required List<SearchSnippet> foundSearchSnippets,
+    required List<EmailId> notFoundEmailIds,
+  }) => {
+    "sessionState": sessionState.value,
+    "methodResponses": [
+      [
+        "Email/query",
+        {
+          "accountId": accountId.id.value,
+          "ids": foundSearchSnippets
+            .map((searchSnippet) => searchSnippet.emailId.id.value)
+            .toList()
+            ..addAll(notFoundEmailIds.map((emailId) => emailId.id.value)),
+        },
+        "c0"
+      ],
+      [
+        "SearchSnippet/get",
+        {
+          "accountId": accountId.id.value,
+          "notFound": notFoundEmailIds.map((emailId) => emailId.id.value).toList(),
+          "state": state.value,
+          "list": foundSearchSnippets.map((searchSnippet) => searchSnippet.toJson()).toList(),
+        },
+        "c1"
+      ]
+    ]
+  };
+
+  Map<String, dynamic> generateRequest({
+    required List<SearchSnippet> foundSearchSnippets,
+    required List<EmailId> notFoundEmailIds,
+    required Filter filter,
+  }) => {
+    "using": [
+      "urn:ietf:params:jmap:core",
+      "urn:ietf:params:jmap:mail"
+    ],
+    "methodCalls": [
+      [
+        "Email/query",
+        {
+          "accountId": accountId.id.value,
+          "filter": filter.toJson(),
+        },
+        "c0"
+      ],
+      [
+        "SearchSnippet/get",
+        {
+          "accountId": accountId.id.value,
+          "filter": filter.toJson(),
+          "#emailIds": {
+            "resultOf": "c0",
+            "name": "Email/query",
+            "path": "/ids/*"
+          },
+        },
+        "c1"
+      ]
+    ]
+  };
+
+  group('search snippet get method test:', () {
+    test(
```

Comment:
can you test the case if searchSnippet error

Reply:
Done